schemas/api/channelChangedEvent.schema.json

```diff
@@ -0,0 +1,53 @@
+{
+	"$schema": "http://json-schema.org/draft-07/schema#",
+	"$id": "https://fdc3.finos.org/schemas/next/api/channelChangedEvent.schema.json",
+	"type": "object",
+	"title": "channelChanged Event",
+	"description": "An event message from the Desktop Agent to an app indicating that its current user channel has changed.",
+	"allOf": [
+		{
+			"$ref": "agentEvent.schema.json"
+		},
+		{
+			"type": "object",
+			"properties": {
+				"type": {
+					"$ref": "#/$defs/ChannelChangedEventType"
+				},
+				"payload": {
+					"$ref": "#/$defs/ChannelChangedEventPayload"
+				},
+				"meta": true
+			},
+			"additionalProperties": false
+		}
+	],
+	"$defs": {
+		"ChannelChangedEventType": {
+			"title": "ChannelChanged Event Message Type",
+			"const": "channelChangedEvent"
+		},
+		"ChannelChangedEventPayload": {
+			"title": "channelChanged Event Payload",
+			"type": "object",
+			"properties": {
+				"newChannelId": {
```

Comment:
It seems like it would be good idea to also pass through the details of all the user channels, in case new user channels have been added or removed.

Reply:
Meh, thats not something that FDC3 provides for currently. If we want to do that we should first add the idea that the user channels can change to the Standard before providing messaging relating to it. It's not currently stated either way (fixed or changeable set) but I've always read it as fixed as I think will most app developers (who may have called getUserChannels() once on startup).

If we do support a changing channel list, that should be its own event as its not likely to change as the user selects a channel and should not only be supported on that event occurring. We'll also need to ensure that channel selector UIs can receive the iframeChannels message again to update themselves...

To summarize, if we support channels changing it should be a separate event (both in FDC3 for the web and the DesktopAgent API, so apps can listen for it) and not only handled through this one. Hence, it should go on a new issue if we want to discuss doing this.